src/System.Net.WebClient/src/System/Net/WebClient.cs

```diff
@@ -875,6 +875,11 @@ private byte[] DownloadBits(WebRequest request, Stream writeStream)
                     }
                     writeStream.SetLength(copyBuffer.Length);
                 }
+                
+                if (contentLength > 0)
```

Comment:
if contentLength == 0 ?

Reply:
It won't actually change any behavior to change this to be `if (contentLength >= 0)`, right?  If so, I suggest we do so, if only for clarity, and so that TotalBytesToReceive more accurately represents the number of bytes we expect, in all cases where we know it.